nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/services/slack/SlackRestService.java

```diff
@@ -0,0 +1,116 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.services.slack;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.commons.io.IOUtils;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.util.StringUtils;
+import org.apache.nifi.web.client.api.HttpResponseEntity;
+import org.apache.nifi.web.client.api.HttpUriBuilder;
+import org.apache.nifi.web.client.provider.api.WebClientServiceProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.OptionalLong;
+
+public class SlackRestService {
+    private final WebClientServiceProvider webClientServiceProvider;
+    private final String accessToken;
+    private final String apiUrl;
+    private final ObjectMapper objectMapper;
+    private final Logger logger;
+
+
+    public SlackRestService(final WebClientServiceProvider webClientServiceProvider,
+                            final String accessToken,
+                            final String apiUrl) {
+        this.webClientServiceProvider = webClientServiceProvider;
+        this.accessToken = accessToken;
+        this.apiUrl = apiUrl;
+        this.objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        this.logger = LoggerFactory.getLogger(SlackRestService.class);
+    }
+
+    public void sendMessageToChannel(final String message, final String channel) throws SlackRestServiceException {
+        final URI apiUri = URI.create(apiUrl);
+        final HttpUriBuilder uriBuilder = webClientServiceProvider.getHttpUriBuilder()
+                .scheme(apiUri.getScheme())
+                .host(apiUri.getHost())
+                .encodedPath(apiUri.getPath())
+                .addPathSegment("chat.postMessage");
+        if (apiUri.getPort() != -1) {
+            uriBuilder.port(apiUri.getPort());
+        }
+        final URI uri = uriBuilder.build();
+
+        final ObjectNode requestBodyJson = objectMapper.createObjectNode();
+        if (StringUtils.isEmpty(channel)) {
+            throw new SlackRestServiceException("The channel must be specified.");
+        }
+        requestBodyJson.put("channel", channel);
+
+        if (StringUtils.isEmpty(message)) {
+            throw new SlackRestServiceException("No message to be sent with this record.");
+        }
+        requestBodyJson.put("text", message);
+
+
+        final InputStream requestBodyInputStream;
+        try {
+            requestBodyInputStream = IOUtils.toInputStream(
```

Comment:
Recommend replacing `IOUtils.toInputStream()` with a simple method that returns `ByteArrayInputStream`. That would allow removing the dependency on `commons-io`.

Reply:
Changing it to ByteArrayInputStream